File: tests/test_rotate_op.cpp
#include <opencv2/opencv.hpp>
#include <string>

#include "common/utils.hpp"
#include "vdsp_ops/rotate_op.hpp"
int main(int argc, char* argv[]) {
  int device_id = 0;
  int rotate_type = atoi(argv[1]);

  vsx::SetDevice(device_id);

  std::string img_path = "../data/images/cycling.jpg";
  cv::Mat img = cv::imread(img_path);

  vsx::RotateOp rotate_op("../samples/ocr_e2e/simple_rotate_debug", device_id);

  vsx::Image vsx_image;
  vsx::MakeVsxImage(img, vsx_image, vsx::ImageFormat::RGB_PLANAR);

  auto result = rotate_op.Process(vsx_image, vsx::rotate_degree_e(rotate_type));

  cv::Mat bgr888;
  vsx::ConvertVsxImageToCvMatBgrPacked(result.Clone(), bgr888);

  cv::imwrite("img_rotated.jpg", bgr888);
  std::cout << "Write result to img_rotated.jpg.\n";
  return 0;
}

File: tests/test_rotate_op.py
from vdsp_ops import RotateOp, RotateDegree
import sys 
import cv2
import os

import vaststreamx as vsx


current_file_path = os.path.dirname(os.path.abspath(__file__))
common_path = os.path.join(current_file_path, "../..")
sys.path.append(common_path)

import common.utils as utils

image = "../../data/images/ppocr.jpg"

device_id = 0

rotate_op = RotateOp(elf_file="./simple_rotate_debug",device_id=device_id)

image_cv = cv2.imread(image)
assert image_cv is not None, f"failed to read image: {image}"
degree_type = int(sys.argv[1])


print(f"rotate degree_type: {degree_type}")

rotate_angle = RotateDegree.ROTATE_DEGREE_90
if degree_type == 0:
    rotate_angle = RotateDegree.ROTATE_DEGREE_90
elif degree_type == 1:
    rotate_angle = RotateDegree.ROTATE_DEGREE_180
elif degree_type == 2:
    rotate_angle = RotateDegree.ROTATE_DEGREE_270
elif degree_type == 4:
    rotate_angle = RotateDegree.ROTATE_DEGREE_NEG90
elif degree_type == 3:
    rotate_angle = RotateDegree.ROTATE_DEGREE_NEG270
else:
    raise Exception(f"unsupported rotate degree_type: {degree_type}")


vsx_image = utils.cv_bgr888_to_vsximage(image_cv, vsx.ImageFormat.BGR_INTERLEAVE, device_id)
print(f"vsx_image shape:{vsx_image.shape}")
org_image = utils.vsximage_to_cv_bgr888(vsx_image)
print(f"org_image shape:{org_image.shape}")

cv2.imwrite("ori.jpg", org_image)

print(f"rotate_angle:{rotate_angle}")
output_vacc = rotate_op(vsx_image, rotate_angle)

print(f"output format:{output_vacc.format}")
cv_output = utils.vsximage_to_cv_bgr888(output_vacc)

cv2.imwrite("rotate_result.jpg",cv_output)

print(f"rotate_result.jpg saved")